runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -123,6 +139,10 @@ public int activeCount() {
     return activeCount.intValue();
   }
 
+  public int maximumThreadCount() {
+    return maximumThreadCount.intValue();
```

Comment:
github won't let me add a comment, but above allThreadsActiveTime shoudl be synchronized so it isn't racy reading that while other threads are modifying it.

can add a @GuardedBy annotation to totalTimeMaxActiveThreadsUsed